### ISSUE TITLE
icons.scss should be built in to 0-18.json somehow

### ISSUE THREAD

**Issue by bbarry** [factoriolab/factorio-lab#87]
in order to more easily support mods eventually, it would be nice if the declarations of icons.scss were built in to the json object...

For example:

```
.chemical-plant {
  width: 32px;
  height: 32px;
  background-position: -32px -672px;
}
```

should be a part of 

```
    {
      "id": "chemical-plant",
      "name": "Chemical Plant",
      "stack": 10,
      "category": "production",
      "row": 5,
      "factory": {
        "speed": 1,
        "modules": 3,
        "electric": 210,
        "drain": 7
      }
    },
```

Once this is done it looks like most non-looping recipe mods (input loops might be harder to solve appropriately) can be accomplished by simply switching out the json and image asset files.

**Comment by dcbroad3** [comment]
The current implementation of this is very mod-unfriendly, but this suggestion would likely measurably affect performance, since Angular would have to calculate the size and position of every icon. 
As a compromise, what if the icon css like this is separated into a simple css file in the assets folder alongside the json and png? It shouldn't be much harder for a mod utility to build the css sheet in addition to the json and sprite png. 
Once I get a bit further I may also move away from using classes directly also, to prevent mod collisions with internal class names, but that's another issue.

**Comment by bbarry** [comment]
Yep I don't have a very good solution for it. Perhaps something like this for icons.scss? 

```
.size-32 {
  width: 32px;
  height: 32px;
}
.size-64 {
  width: 64px;
  height: 64px;
}

@for $i from 0 through 10 {
  @for $j from 0 through 50 {
  .index-#{$i}-#{$j} {
    background-position: -#{$i * 32}px -#{$j * 32}px;
    }
  }
}
```

Then (a bunch of work to put the right classes in this file).

```
 {
      "id": "chemical-plant",
      "name": "Chemical Plant",
      "stack": 10,
      "category": "production",
      "row": 5,
      "factory": {
        "speed": 1,
        "modules": 3,
        "electric": 210,
        "drain": 7
      },
      "icon": "size-32 index-1-21"
    },

```

Surely someone can come up with something better. Is there anything similar in the lua configuration files that could be leveraged?

**Comment by dcbroad3** [comment]
Hey @bbarry, looks like you saw my PR, but it's merged now and this should be resolved, let me know if you think the update didn't fully resolve this issue.

**Comment by bbarry** [comment]
I think that looks great and it seems fine performance wise in Firefox, Chrome and IE11 (there are a few bugs in IE11 in the settings tab though; not sure you should care about it). Sorry I couldn't be more helpful on this; I am still trying to get my machine on a more reliable internet that can actually use npm and clone some of these repositories that I want to contribute to.

**Comment by dcbroad3** [comment]
> there are a few bugs in IE11 in the settings tab though; not sure you should care about it

I really only intend to support Chromium (Chrome/Edge) and Firefox browsers unless there's a real push to support others.

Thanks for your comments!
